src/audio/copier/copier.c

```diff
@@ -946,12 +946,14 @@ static int copier_reset(struct comp_dev *dev)
 	cd->input_total_data_processed = 0;
 	cd->output_total_data_processed = 0;
 
+	if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
+		if (cd->hd->chan)
+			notifier_unregister(dev,
+					    cd->hd->chan, NOTIFIER_ID_DMA_COPY);
+	}
+
 	for (i = 0; i < cd->endpoint_num; i++) {
 		if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
-			if (cd->hd->chan)
```

Comment:
by the same argument in one of your previous patches that endpoint_num for host copier is never greater than 1, we can keep the notifier_unregister within the loop as well no?

Reply:
yes, this will make code looks much better, and move out when endpoint num > 1.

Reply:
ok, I added one more flag for double safe.